app/controllers/leaderboard.js

```diff
@@ -25,309 +24,230 @@ module.exports.syncAthletes = function(req, res, next) {
+      if (athlete.wod1Rank == -1 || athlete.wod2Rank == -1 || athlete.wod3Rank == -1 || athlete.wod4Rank == -1) {
+        console.log("Athlete with overallScore not updated:");
+        console.log(athlete);
       } else {
-        next++;
-        prevRank = pos;
-        pos = next;
-      }
-
-      prev = athlete.wod3Score;
-      prevTieBreak = athlete.wod3TieBreak;
+        Athlete.update({id: athlete.id}, {$set: {overallScore: _overallScore}}, function(err, result) {
+          if (err) console.log(err);
 
-      Athlete.update({id: athlete.id}, {$set: {wod3Rank: rank}}, function(err, result) {
-        if (err) console.log(err);
+          console.log('Overall score updated for ' + athlete.name);
+          console.log('score ' + athlete.overallScore);
+        });
+      }
 
-        console.log('Updated rank of ' + athlete.name);
-        callback();
-      });
+      callback();
     });
   });
 
-  // /* Men wod4 */
-  Athlete.find({sex: 'm'}).sort({wod4Score: -1, wod4TieBreak: 1}).exec(function(err, athletes) {
-    if (err) console.log(err);
-
-    var pos = 1, next = 1, prev = 0, prevTieBreak = 0, prevRank = 1;
-    async.eachSeries(athletes, function(athlete, callback) {
-      var rank = pos;
-
-      if (athlete.wod4Score == prev && athlete.wod4TieBreak == prevTieBreak) {
-        rank = prevRank;
-      } else {
-        next++;
-        prevRank = pos;
-        pos = next;
-      }
+  res.sendStatus(200);
+};
 
-      prev = athlete.wod4Score;
-      prevTieBreak = athlete.wod4TieBreak;
+module.exports.generate = function(req, res, next) {
+  generateJSON();
+  res.sendStatus(200);
+};
 
-      Athlete.update({id: athlete.id}, {$set: {wod4Rank: rank}}, function(err, result) {
-        if (err) console.log(err);
+/* protected */
 
-        console.log('Updated rank of ' + athlete.name);
-        callback();
-      });
-    });
-  });
+function saveAthletesResults(rx, scale, not_performed, sort, score, tiebreak, rank) {
+  var allAthletes = [];
 
-  /* Women wod1 */
-  Athlete.find({sex: 'f'}).sort({wod1Score: -1, wod1TieBreak: 1}).exec(function(err, athletes) {
+  Athlete.find(rx).sort(sort).exec(function(err, athletes) {
     if (err) console.log(err);
 
-    var pos = 1, next = 1, prev = 0, prevTieBreak = 0, prevRank = 1;
-    async.eachSeries(athletes, function(athlete, callback) {
-      var rank = pos;
+    allAthletes = allAthletes.concat(athletes);
+  }).then(function() {
+    Athlete.find(scale).sort(sort).exec(function(err, athletes) {
+      if (err) console.log(err);
 
-      if (athlete.wod1Score == prev && athlete.wod1TieBreak == prevTieBreak) {
-        rank = prevRank;
-      } else {
-        next++;
-        prevRank = pos;
-        pos = next;
-      }
-
-      prev = athlete.wod1Score;
-      prevTieBreak = athlete.wod1TieBreak;
-
-      Athlete.update({id: athlete.id}, {$set: {wod1Rank: rank}}, function(err, result) {
+      allAthletes = allAthletes.concat(athletes);
+    }).then(function() {
```

Comment:
To avoid so many `.then()` and create the callback hell, we use a technique that you just return the statement instead of use the `.then()` of this method. Let me show an example with callback hell that is your case:

```
method.then(function() {
  another.then(function() {
    moreone.then(function() {
    });
  });
});
```
To avoid it, you just use the `return` statement:
```
method.then(function() {
  return another();
}).then(function() {
  return moreone();
}).then(function() {
});
```